### ISSUE TITLE
Unknown MySQL server error

### ISSUE THREAD

**Issue by fuhton** [philipnewcomer/wpdocker#78]
Following the [installation instructions](https://github.com/Pilothouse-App/Pilothouse/wiki/Installation), I've attempted to create a new WordPress instance. After I've filled out all the options, I get the following Error - `ERROR 2005 (HY000): Unknown MySQL server host 'mysql' (-2)`. I've included all my responses to the prompts below. Let me know if there's something I missed from the instructions.

Prompt | Response
------------ | -------------
Type | WordPress
PHP version | Global default
Local Domain | default
URL to proxy | (left blank)
Git repo for wp-content | (left blank)

**Comment by philipnewcomer** [comment]
At what point are you seeing the error, is it while the `create` command is running, or during an action after it is completed? Can you post the entire output from the command?

**Comment by fuhton** [comment]
```
~ nicholas$ pilothouse create nicks_site
? Type of site: WordPress
? PHP version: Global default
? Local domain name to use: nicks_site.dev
? URL to proxy uploads from: 
? Git repository to clone for the wp-content directory: 
ERROR 2005 (HY000): Unknown MySQL server host 'mysql' (-2)
```

**Comment by philipnewcomer** [comment]
Open Kitematic (It should have been installed along with Docker) and verify that the `mysql` container is running. Also, just to double check, you did run `pilothouse up` first, didn't you?

**Comment by fuhton** [comment]
So this error occurs during `pilothouse up`. The `mysql` container errors out pretty quickly.

```
2017-10-01 13:25:04 139812629264256 [Note] mysqld (mysqld 10.2.9-MariaDB-10.2.9+maria~jessie) starting as process 1 ...
2017-10-01 13:25:04 139812629264256 [Warning] InnoDB: Using innodb_file_format is deprecated and the parameter may be removed in future releases. See http://dev.mysql.com/doc/refman/5.7/en/innodb-file-format.html
2017-10-01 13:25:04 139812629264256 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2017-10-01 13:25:04 139812629264256 [Note] InnoDB: Uses event mutexes
2017-10-01 13:25:04 139812629264256 [Note] InnoDB: Compressed tables use zlib 1.2.8
2017-10-01 13:25:04 139812629264256 [Note] InnoDB: Using Linux native AIO
2017-10-01 13:25:04 139812629264256 [Note] InnoDB: Number of pools: 1
2017-10-01 13:25:04 139812629264256 [Note] InnoDB: Using SSE2 crc32 instructions
2017-10-01 13:25:04 139812629264256 [Note] InnoDB: Initializing buffer pool, total size = 2G, instances = 8, chunk size = 128M
2017-10-01 13:25:04 139812629264256 [Note] InnoDB: Completed initialization of buffer pool
2017-10-01 13:25:04 139809916053248 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2017-10-01 13:25:04 139812629264256 [ERROR] InnoDB: Header page consists of zero bytes in datafile: ./ibdata1, Space ID:0, Flags: 0. Please refer to http://dev.mysql.com/doc/refman/5.7/en/innodb-troubleshooting-datadict.html for how to resolve the issue.
2017-10-01 13:25:04 139812629264256 [ERROR] InnoDB: Corrupted page [page id: space=0, page number=0] of datafile './ibdata1' could not be found in the doublewrite buffer.
2017-10-01 13:25:04 139812629264256 [ERROR] InnoDB: Plugin initialization aborted with error Data structure corruption
2017-10-01 13:25:05 139812629264256 [Note] InnoDB: Starting shutdown...
2017-10-01 13:25:05 139812629264256 [ERROR] Plugin 'InnoDB' init function returned error.
2017-10-01 13:25:05 139812629264256 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2017-10-01 13:25:05 139812629264256 [Note] Plugin 'FEEDBACK' is disabled.
2017-10-01 13:25:05 139812629264256 [ERROR] Could not open mysql.plugin table. Some plugins may be not loaded
2017-10-01 13:25:05 139812629264256 [ERROR] Unknown/unsupported storage engine: InnoDB
2017-10-01 13:25:05 139812629264256 [ERROR] Aborting```

**Comment by fuhton** [comment]
I've tracked down the issue to an assumption that `mysql` is installed in a specific path `/var/lib/mysql`. I've forked and [pushed up a fix](https://github.com/fuhton/Pilothouse/commit/0ebc4b6ddf113b85cdd839c35d845414ef3b7f7e) that works for my homebrew installed version of mysql. Is there documentation or settings that allow overwriting of this path directly?

**Comment by philipnewcomer** [comment]
Hmmm, so it sounds like you were attempting to use a MySQL server running on your host OS instead of using the `mysql` container provided by Pilothouse? Can you describe what the use case is for doing this and why the Docker MySQL container that comes with PH doesn't suit your needs?

Although I'm certainly not opposed to making Pilothouse more configurable, the goal of Pilothouse is to have all services (including MySQL) be provided by a Docker container, and I'm not sure I want to provide support for having one or more of the core services running outside of the Docker stack on the host.

**Comment by fuhton** [comment]
So reviewing everything and the changes I made, I think I had originally downloaded a corrupted mysql volume. Adding in my local copy as a volume somehow has righted everything. When using the develop branch of this repo, I've confirmed that the mysql volumes used are the original and do not include my local install of mysql and that mysql initializes successfully. I'm not quite satisfied with how this has been "fixed", but it has resolved the original issue I submitted here. Thank you for all your thoughtful responses!
